File: frontend/libs/aas-designer-core/src/lib/organisation/model/apikey.ts
import { Organisation } from '@aas-designer-model';
import { Benutzer } from '@aas-designer-model';

export class Apikey {
  id: number | null = null;
  key: string = '';
  notice: string = '';
  active: boolean = false;
  validUntil: Date | null = null;
  scopes: string[] = [];

  organisation: Organisation | null = null;
  organisationId: number | null = null;

  benutzer: Benutzer | null = null;
  benutzerId: number | null = null;

  static fromDto(dto: any) {
    const token = new Apikey();
    Object.assign(token, dto);
    if (dto.validUntil != null) token.validUntil = new Date(dto.validUntil);
    if (dto.benutzer != null) token.benutzer = Benutzer.fromDto(dto.benutzer);
    if (dto.organisation != null) token.organisation = Organisation.fromDto(dto.organisation);

    return token;
  }
}

File: services/shared/AasShared/Model/Apikey.cs
using Newtonsoft.Json;

namespace AasShared.Model;

public class Apikey
{
    public long Id { get; set; }
    public Guid Key { get; set; } = Guid.Empty;
    public DateTime? ValidUntil { get; set; }
    public bool Active { get; set; }
    public string Notice { get; set; } = string.Empty;
    public List<string> Scopes { get; set; } = [];
    public long? OrganisationId { get; set; }
    public long? AasInfrastructureSettingsId { get; set; }
    public long? BenutzerId { get; set; }
}
